build.ps1

```diff
@@ -103,26 +103,25 @@ try {
 
     if ($run)
     {
-        # Run cleanup first so the samples don't get overloaded with old device instances
+        # Run cleanup first so the samples don't get overloaded with old devices
         RunApp provisioning\Samples\service\CleanupEnrollmentsSample "Provisioning\Service\CleanupEnrollmentsSample"
         RunApp iot-hub\Samples\service\CleanUpDevicesSample "IoTHub\Service\CleanUpDevicesSample"
 
-        RunApp iot-hub\Samples\device\FileUploadSample "IoTHub\Device\FileUploadSample"
-        RunApp iot-hub\Samples\device\KeysRolloverSample "IoTHub\Device\KeysRolloverSample"
-        RunApp iot-hub\Samples\device\MessageSample "IoTHub\Device\MessageSample"
-        RunApp iot-hub\Samples\device\MethodSample "IoTHub\Device\MethodSample"
-        RunApp iot-hub\Samples\device\TwinSample "IoTHub\Device\TwinSample"
-
-        RunApp iot-hub\Samples\module\MessageSample "IoTHub\Module\MessageSample"
-        RunApp iot-hub\Samples\module\TwinSample "IoTHub\Module\TwinSample"
+        RunApp iot-hub\Samples\device\FileUploadSample "IoTHub\Device\FileUploadSample" "-p$env:IOTHUB_DEVICE_CONN_STRING"

```

Comment:
I got a weird error in powershell where sending the params as "-p $env:IOTHUB_DEVICE_CONN_STRING" introduced an extra whitespace before the value of `$env:IOTHUB_DEVICE_CONN_STRING`. Adding braces/ quotes around `$env:IOTHUB_DEVICE_CONN_STRING` didn't help either.

The only thing that worked was to send "-p$env:IOTHUB_DEVICE_CONN_STRING" (without the whitespace delimiter), but this seems inherently wrong to me.

Any ideas?

Reply:
Yeah, this is going to look like a mistake to any future observer.

Have you tried `$($env.IOTHUB_DEVICE_CONN_STRING)`? You might have to do that if encasing it in quotes, anyway. That is how you embed a hierarchical variable in a string in powershell.

If nothing else works, then put a comment in to explain why there is no whitespace.

Reply:
`RunApp iot-hub\Samples\device\FileUploadSample "IoTHub\Device\FileUploadSample" "-p $($env:IOTHUB_DEVICE_CONN_STRING)"` also introduced the space 😞 
I'll add a note explaining this workaround for now.

@azabbasi - you can update it with your findings when you have the time :)

Reply:
@drwill-ms ,@azabbasi - This is now solved, just FYI

Reply:
double quotes eh?

Reply:
Actually it was related to the way the command was being invoked. Invoking with `&` gives rise to this issue, whereas invoking with `Invoke-Expression` works as expected.